### DISCUSSION TITLE
Uno Templates 4.7.37.28 and CommunityToolkit does not build

### DISCUSSION THREAD

**Discussion by corvand** [unoplatform/uno#11363] [Q&A] [answered]
Hi, having an issue building Uno trying to use the community toolkit have the following reproduction scenario:

Uno Templates 4.7.37.28

Vs2022 17.4.5

Create Uno Platform App select:
Desktop WinUI(Windows App SDK) and 
Gtk(Linux, macOS, Windows 7 or later)

Build the app, run uno-check everything works fine.

Now add the Community Toolkit

CommunityToolkit.Winui.UI.Controls on App1 and Platform\App1.Windows

And Add Uno.CommunityToolkit.Winui.UI.Controls on Platform\App1.Gtk

On Mainpage.xaml add:

xmlns:controls="using:CommunityToolkit.WinUI.UI.Controls" as reference and

Add in the Grid:

<controls:RadialGauge></controls:RadialGauge>

Building this results in:

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0400	The type or namespace name 'CommunityToolkit' could not be found in the global namespace (are you missing an assembly reference?)	App1 (net7.0-windows10.0.18362)	F:\Projects\Repos\App1\App1\App1\Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MainPage_b88362a9453e05dfd24cf37e9a83540f.cs	94	Active

**Comment by jeromelaban** [comment, marked as answer]
Thanks for the question. Have you added [these lines](https://platform.uno/docs/articles/uno-community-toolkit.html?tabs=tabid-winui#referencing-the-windows-community-toolkit-from-a-cross-targeted-library) in the `App1.csproj`? (the conditional block)

**Reply by corvand** [reply]
No because the structure is now different possibly it has something to do with App1 referencing net7.0 and net7.0-windows10.0.18362
the first one has a strange empty toolkit structure.
![image](https://user-images.githubusercontent.com/10348392/219708217-0ab6e762-e1c8-4999-8698-5ae217eb05b1.png)

<ItemGroup>
    <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="5.0.0" />
    <PackageReference Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
    <PackageReference Include="Uno.WinUI" Version="4.7.37" />
  </ItemGroup>
  <ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.18362'">
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.1.3" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.22000.24" />
    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" TargetingPackVersion="10.0.22000.24" />
  </ItemGroup>

**Reply by jeromelaban** [reply]
Why did you not add the same lines as specified in the documentation? What makes you think it is different?

**Reply by corvand** [reply]
In the former template version i could configure it by putting the uno.community version on the Gtk project and the community on the Win project avoiding the if structure. Now in the new template model that is no longer using the app.shared model i hounestly dont' now how to do that, i tried but it is comlpetley failing if alter the file.... my guess is that the double net dependency is messing things up see also that the drildown on the net7.o for the community framework is absent, and if you look in the mainpage.xaml and switch the framework, it can no longer parse even the grid, i will atach the screenshots..

**Reply by jeromelaban** [reply]
To add the depedency to the class library project, you'll need to add the following block:

```xml
<ItemGroup Condition="'$(TargetFramework)' == 'net7.0-windows10.0.18362.0'">
  <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
</ItemGroup>
<ItemGroup Condition="'$(TargetFramework)' != 'net7.0-windows10.0.18362.0'">
  <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.100" />
</ItemGroup>
```

Is this what you tried ?

**Reply by corvand** [reply]
If i do that:

Severity	Code	Description	Project	File	Line	Suppression State
Error		The Uno Platform version of Community Toolkit is not supported on WinAppSDK. Use the official packages instead, see https://aka.platform.uno/install-uno-community-toolkit for details.	App1.Windows	C:\Users\Cor\.nuget\packages\uno.communitytoolkit.winui.ui\7.1.100\buildTransitive\net5.0-windows10.0.18362.0\uno.communitytoolkit.winui.ui.targets	6

**Reply by corvand** [reply]
could it be that a nuget package cannot be referenced by two .net packages ?

**Reply by jeromelaban** [reply]
Make sure that `net7.0-windows10.0.18362.0` matches the target framework specified at the top of the project.

**Reply by jeromelaban** [reply]
Actually, let's use this, so that you do not have to match versions:

```xml
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">
  <PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
</ItemGroup>
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">
  <PackageReference Include="Uno.CommunityToolkit.WinUI.UI.Controls" Version="7.1.100" />
</ItemGroup>
```

**Reply by corvand** [reply]
Yep that does the trick!!

**Reply by jeromelaban** [reply]
Thanks! We'll update our documentation.

**Comment by corvand** [comment]
Thank you, was stuck on this issue for a couple of days, can continue to migrate all of our windows iot for raspberry apps now towards linux and stil maintain our xaml graph sources.

Verzonden vanaf Outlook voor Android<https://aka.ms/AAb9ysg>
________________________________
